### PR TITLE
Add support for multiline toplevel comments in .vue files

### LINKED ISSUE #957: Multiline toplevel HTML Comments in .vue files are not supported
**I want to report a bug.**

**SonarQube version:** Version 7.0 (build 36138)<br>

**Reproducer** 
```
<!-- Test comment
comment
-->
<template>
  <div></div>
</template>

<script>
</script>

<style>
</style>
```

**Logs**
```
ERROR: Unable to parse file: file:///C:/Users/...../test.vue

ERROR: Parse error at line 2 column 1:

 1: <!-- Test comment
 2: comment
    ^
 3: -->
 4: <template>
 5:   <div></div>
 6: </template>
 7:
 8: <script>
 9: </script>
10:
11: <style>
12: </style>
```

**Expected behavior**

According to the [Vue Component Spec](https://vue-loader.vuejs.org/en/start/spec.html), this is a valid .vue file, as this is valid HTML comment syntax. Vue files with single-line toplevel HTML comments work as expected.

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**
Fixes #957

### CODE REVIEW

File: javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java [hunk 1]
```diff
@@ -455,12 +459,20 @@ private static void lexical(LexerlessGrammarBuilder b) {
...
+    b.rule(VUE_SPACING).is(
+      EMPTY_LINE,
+      b.zeroOrMore(
+        VUE_COMMENT,
+        EMPTY_LINE))
+      .skip();
+
+    b.rule(SCRIPT_TAG).is(VUE_SPACING, b.token(GenericTokenType.IDENTIFIER, "<script"));
+    b.rule(SCRIPT_SECTION_TS).is(VUE_SPACING, b.regexp("(?s)<script\\s+lang=\"ts\"\\s*>.*</script>"));
+    b.rule(SCRIPT_TAG_CLOSE).is(VUE_SPACING, b.token(GenericTokenType.IDENTIFIER, "</script>"));
```
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java]: here it should be simple `SPACING` (do we parse `<script> /*foo*/</script>`

File: javascript-frontend/src/test/java/org/sonar/javascript/parser/declarations/module/ScriptTest.java [hunk 2]
```diff
@@ -69,6 +70,10 @@ public void vue_file() throws Exception {
       .matches("<script>var i;</script>" + styleWithCss + componentTemplate)
       .matches(styleWithCss + "<script>var i;</script>" + componentTemplate)
       .matches(styleWithCss + componentTemplate + "<script>var i;</script>")
+
+      .matches(multiLineComment + "<script>var i;</script>")
```
Comment (vilchik-elena) [inline javascript-frontend/src/test/java/org/sonar/javascript/parser/declarations/module/ScriptTest.java]: IMO we need more tests here (multiple comments, comments in front of each vue part, comments without new lines, with empty lines)

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/JavaScriptSensorTest.java [hunk 3]
```diff
@@ -347,6 +347,12 @@ public void should_analyse_vue_with_custom_sections() {
     assertThat(context.allIssues()).hasSize(1);
   }
 
+  @Test
+  public void should_analyse_vue_with_multiline_comment() {
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/JavaScriptSensorTest.java]: I don't see value of this test, do you?

Comment (andrea-guarino-sonarsource) [reply sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/JavaScriptSensorTest.java]: I agree that it mainly overlaps with the other unit tests. I will remove it.

File: javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java [hunk 4]
```diff
@@ -469,7 +469,7 @@ private static void lexical(LexerlessGrammarBuilder b) {
 
     b.rule(SCRIPT_TAG).is(VUE_SPACING, b.token(GenericTokenType.IDENTIFIER, "<script"));
     b.rule(SCRIPT_SECTION_TS).is(VUE_SPACING, b.regexp("(?s)<script\\s+lang=\"ts\"\\s*>.*</script>"));
-    b.rule(SCRIPT_TAG_CLOSE).is(VUE_SPACING, b.token(GenericTokenType.IDENTIFIER, "</script>"));
+    b.rule(SCRIPT_TAG_CLOSE).is(SPACING, b.token(GenericTokenType.IDENTIFIER, "</script>"));
```
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/parser/EcmaScriptLexer.java]: no test for this case
